components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+    #[allow(dead_code)]
+    fn long_marheshvan(h_year: i32) -> bool {
+        let coll = [355, 385];
+        coll.contains(&Self::days_in_hebrew_year(h_year))
+    }
+    // True if the month Kislve is going to be short in given Hebrew year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2326
+    #[allow(dead_code)]
+    fn short_kislev(h_year: i32) -> bool {
+        let coll = [353, 383];
+        coll.contains(&Self::days_in_hebrew_year(h_year))
+    }
+
+    // Last day of month (h_month) in Hebrew year (h_year)
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2230
+    #[allow(dead_code)]
+    fn last_day_of_hebrew_month(h_year: i32, h_month: u8) -> u8 {
+        match h_month {
+            IYYAR | TAMMUZ | ELUL | TEVET | ADARII => 29,
+            ADAR => {
+                if !Self::is_leap_year(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            MARHESHVAN => {
+                if !Self::long_marheshvan(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            KISLEV => {
+                if Self::short_kislev(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            _ => 30,
+        }
+    }
+
+    // "Fixed" is a day count representation of calendars staring from Jan 1st of year 1 of the Georgian Calendar.
+    // The fixed date algorithms are from
+    // Dershowitz, Nachum, and Edward M. Reingold. _Calendrical calculations_. Cambridge University Press, 2008.
+    //
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2331
+    #[allow(dead_code)]
+    fn fixed_from_hebrew(date: HebrewDateInner) -> RataDie {
+        let year = date.0.year;
+        let month = date.0.month;
+        let day = date.0.day;
+
+        // Days so far this month.
+        let mut total_days = Self::hebrew_new_year(year) + day.into() - 1;
+
+        if month < TISHRI {
+            // Then add days in prior months this year before
+            for m in (TISHRI..=Self::last_month_of_hebrew_year(year)).chain(NISAN..month) {
+                total_days += Self::last_day_of_hebrew_month(year, m).into();
+            }
+        } else {
+            // Else add days in prior months this year
+            for m in TISHRI..month {
+                total_days += Self::last_day_of_hebrew_month(year, m).into();
+            }
+        }
+
+        total_days
+    }
+
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2352
+    #[allow(dead_code, clippy::unwrap_used)]
+    fn hebrew_from_fixed(date: RataDie) -> Date<Hebrew> {
+        let approx = helpers::i64_to_i32(
```

Comment:
Note: this isn't exactly the right saturation logic, but we'll fix it in https://github.com/unicode-org/icu4x/issues/3710

Reply:
Okay! :D